src/App.ts

```diff
@@ -237,13 +237,21 @@ export default class App {
       this.logger.setLevel(this.logLevel);
     }
     this.errorHandler = defaultErrorHandler(this.logger);
-    this.clientOptions = {
-      agent,
-      // App propagates only the log level to WebClient as WebClient has its own logger
-      logLevel: this.logger.getLevel(),
-      tls: clientTls,
-      slackApiUrl: clientOptions !== undefined ? clientOptions.slackApiUrl : undefined,
-    };
+
+    this.clientOptions = clientOptions !== undefined ? clientOptions : {};
+    if (agent !== undefined && this.clientOptions.agent === undefined) {
+      this.clientOptions.agent = agent;
+    }
+    if (clientTls !== undefined && this.clientOptions.tls === undefined) {
+      this.clientOptions.tls = clientTls;
+    }
+    if (logLevel !== undefined && logger === undefined) {
+      // If the constructor has both, logLevel is ignored
```

Comment:
I don't think this can affect many existing apps but this is for keeping backward compatibility (just in case).

Reply:
Should we warn the developer in this case in case there's confusion?

Reply:
It sounds nice 👍

Reply:
done